### PR TITLE
Add fast validator for Callable trait

### LINKED ISSUE #718: Fast validator for Callable trait
The Callable trait is currently implemented with a pure Python validate method (using the `callable` check, together with optionally allowing None.  This is a fairly commonly used trait.

The Python C API provides the `PyCallable_check` function, so it would be straightforward to provide a C-level validator.

### PR DESCRIPTION
**Pull request by kitchoi**
Closes #718

### CODE REVIEW

File: traits/tests/test_callable.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        self.assertIs(a.value, func)
+
+    def test_accepts_type(self):
+        MyCallable(value=float)
+
+    def test_accepts_method(self):
+        dummy = Dummy()
+        MyCallable(value=dummy.instance_method)
+
+    def test_accepts_none(self):
```
Comment (mdickinson) [inline traits/tests/test_callable.py]: Would it be worth adding a test that if `allow_none=False` (or is unspecified), then `None` is *not* accepted?

File: traits/ctraits.c [hunk 2]
```diff
@@ -3466,6 +3466,28 @@ validate_trait_function(
...
++----------------------------------------------------------------------------*/
+
+static PyObject *
+validate_trait_callable(
+    trait_object *trait, has_traits_object *obj, PyObject *name,
+    PyObject *value)
+{
+
+    PyObject *type_info = trait->py_validate;
+    int kind = PyTuple_GET_SIZE(type_info);
```
Comment (mdickinson) [inline traits/ctraits.c]: What's the purpose of the `kind` check? Isn't the size always 1 at this point?

I don't think I understand how this works if `allow_none=False` in the original trait.

File: traits/tests/test_callable.py (hunk 1 above)
Comment (kitchoi) [reply traits/tests/test_callable.py]: I added that in the first commit, and that did not pass, indicating that this feature was not supported on master?  So I removed it in the second commit.
```
======================================================================
FAIL: test_rejects_none (traits.tests.test_callable.TestCallable)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/kit/git/traits/traits/tests/test_callable.py", line 78, in test_rejects_none
    MyCallableCannotBeNone(value=None)
AssertionError: TraitError not raised

```

File: traits/ctraits.c (hunk 2 above)
Comment (kitchoi) [reply traits/ctraits.c]: Looks like `allow_none` is not supported on master, or did I do something stupid? It is true that checking the kind here is a bit pointless here.

Comment (mdickinson) [reply traits/ctraits.c]: Agreed; looks like there's no support for `Callable(allow_none=True)` on master. So we should probably remove the `MyCallableCanBeNone` and related tests.

Comment (mdickinson) [reply traits/ctraits.c]: Having said that, I could see `allow_none=False` being potentially useful, but I think that's another issue/PR.

Comment (kitchoi) [reply traits/ctraits.c]: Removing the related tests sound good to me. They are a bit confusing given the feature hasn't been supported.
:+1: to extending `allow_none` support to `Callable`, in a separate issue/PR.

File: traits/trait_types.py [hunk 3]
```diff
@@ -903,6 +903,9 @@ class Callable(TraitType):
     #: A description of the type of value this trait accepts:
     info_text = "a callable value"
 
+    #: The C-level fast validator to use
+    fast_validate = (ValidateTrait.callable, )
```
Comment (mdickinson) [inline traits/trait_types.py]: nit:
```suggestion
    fast_validate = (ValidateTrait.callable,)
```

File: traits/ctraits.c [hunk 4]
```diff
@@ -3466,6 +3466,25 @@ validate_trait_function(
...
+|  Verifies a Python value is a callable (or None):
++----------------------------------------------------------------------------*/
+
+static PyObject *
+validate_trait_callable(
+    trait_object *trait, has_traits_object *obj, PyObject *name,
+    PyObject *value)
+{
+
+    if ((value == Py_None) || (PyCallable_Check(value)> 0)) {
```
Comment (mdickinson) [inline traits/ctraits.c]: nit:
```suggestion
    if ((value == Py_None) || PyCallable_Check(value)) {
```

File: traits/tests/test_callable.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+import unittest
+
+from traits.api import Callable, HasTraits, TraitError
+
+
+
```
Comment (mdickinson) [inline traits/tests/test_callable.py]: nit: too many blank lines

Comment (kitchoi) [reply traits/tests/test_callable.py]: Whoops, I am sorry. My IDE seems to have bailed out on linting.

Comment (kitchoi) [reply traits/tests/test_callable.py]: > My IDE seems to have bailed out on linting.

Ah... it is setup.cfg.  I see!
(Fixed anyway)
